x/emissions/metrics/metrics_util.go

```diff
@@ -0,0 +1,199 @@
+// Metric Names:
+//
+//	allora_deliver_batch_tx_duration_miliseconds
+//	allora_deliver_batch_tx_duration_miliseconds_count
+//	allora_deliver_batch_tx_duration_miliseconds_sum
+func MeasureDeliverBatchTxDuration(start time.Time) {
+	metrics.MeasureSince(
+		[]string{"allora", "deliver", "batch", "tx", "milliseconds"},
+		start.UTC(),
+	)
+}
+
+// allora_epoch_new
+func SetEpochNew(epochNum uint64) {
+	metrics.SetGauge(
+		[]string{"allora", "epoch", "new"},
+		float32(epochNum),
+	)
+}
+
+// Measures number of new websocket connects
+// Metric Name:
+//
+//	allora_websocket_connect
+func IncWebsocketConnects() {
+	telemetry.IncrCounterWithLabels(
+		[]string{"allora", "websocket", "connect"},
+		1,
+		nil,
+	)
+}
+
+// Measures throughput per message type
+// Metric Name:
+//
+//	allora_throughput_<metric_name>
+func SetThroughputMetricByType(metricName string, value float32, msgType string) {
+	telemetry.SetGaugeWithLabels(
+		[]string{"allora", "loadtest", "tps", metricName},
+		value,
+		[]metrics.Label{telemetry.NewLabel("msg_type", msgType)},
+	)
+}
+
+// Measures the number of times the total block gas wanted in the proposal exceeds the max
+// Metric Name:
+//
+//	allora_failed_total_gas_wanted_check
+func IncrFailedTotalGasWantedCheck(proposer string) {
+	telemetry.IncrCounterWithLabels(
+		[]string{"allora", "failed", "total", "gas", "wanted", "check"},
+		1,
+		[]metrics.Label{telemetry.NewLabel("proposer", proposer)},
+	)
+}
+
+// Measures the number of times the total block gas wanted in the proposal exceeds the max
+// Metric Name:
+//
+//	allora_failed_total_gas_wanted_check
+func IncrValidatorSlashed(proposer string) {
+	telemetry.IncrCounterWithLabels(
+		[]string{"allora", "failed", "total", "gas", "wanted", "check"},
+		1,
+		[]metrics.Label{telemetry.NewLabel("proposer", proposer)},
+	)
+}
+
+// Measures the number of times the total block gas wanted in the proposal exceeds the max
+// Metric Name:
+//
+//	allora_tx_gas_counter
+func IncrGasCounter(gasType string, value int64) {
+	telemetry.IncrCounterWithLabels(
+		[]string{"allora", "tx", "gas", "counter"},
+		float32(value),
+		[]metrics.Label{telemetry.NewLabel("type", gasType)},
+	)
+}
+
+// Measures RPC endpoint request throughput
+// Metric Name:
+//
+//	allora_rpc_request_counter
+func IncrementRpcRequestCounter(endpoint string, connectionType string, success bool) {
+	telemetry.IncrCounterWithLabels(
+		[]string{"allora", "rpc", "request", "counter"},
+		float32(1),
+		[]metrics.Label{
+			telemetry.NewLabel("endpoint", endpoint),
+			telemetry.NewLabel("connection", connectionType),
+			telemetry.NewLabel("success", strconv.FormatBool(success)),
+		},
+	)
+}
+
+// Measures the RPC request latency in milliseconds
+// Metric Name:
+//
+//	allora_rpc_request_latency_ms
+func MeasureRpcRequestLatency(endpoint string, connectionType string, startTime time.Time) {
+	metrics.MeasureSinceWithLabels(
+		[]string{"allora", "rpc", "request", "latency_ms"},
```

Comment:
for all these functions, if `connectionType` is kept, would you not be passing it into `[]string{...}` instead of  `"rpc"` ?